Improve tooltip text format

MangelMaxime opened the pull request:
This PR is related to my current work to make the tooltip in Ionide consistant:
 
- https://github.com/ionide/ionide-vscode-fsharp/issues/852
- https://github.com/ionide/ionide-vscode-fsharp/pull/855

### Introduction

The new text generated for the tooltip is now using Markdown synthax. I know that FSAC is used by others project that Ionide so we need to ask if this is the good choice.

Also, I add some section header in the output so it's easier to read and more consistant. Before this PR, the tooltip where already containing some markdown synthax like ``` `source code format` ```

If having the section in Markdown format is a problem we can still replace `**Parameters**` by `Parameters:` and replace it from Ionide side but it would need an extra step.

## Changes

In order to compare before and after, I will use `List.find` to get the tooltip info.

**Source code**

```fs
/// <summary>Returns the first element for which the given function returns True.
/// Raises <c>KeyNotFoundException</c> if no such element exists.</summary>
/// <param name="predicate">The function to test the input elements.</param>
/// <param name="list">The input list.</param>
/// <exception cref="System.Collections.Generic.KeyNotFoundException">Thrown if the predicate evaluates to false for
/// all the elements of the list.</exception>
/// <returns>The first element that satisfies the predicate.</returns>
[<CompiledName("Find")>]
val find: predicate:('T -> bool) -> list:'T list -> 'T
```

### Text generated by FSAC:

**Before**

```markdown
Returns the first element for which the given function returns True.
 Raises <c>KeyNotFoundException</c> if no such element exists.

`list`: The input list.
`predicate`: The function to test the input elements.

Exceptions:
	`System.Collections.Generic.KeyNotFoundException`: Thrown if the predicate evaluates to false for
 all the elements of the list.

Generic parameters:
`'T` is `'b`
```

**Now**

```markdown
Returns the first element for which the given function returns True.
 Raises <c>KeyNotFoundException</c> if no such element exists.

**Parameters**
  * `list`: The input list.
  * `predicate`: The function to test the input elements.

**Exceptions**
  * `System.Collections.Generic.KeyNotFoundException`: Thrown if the predicate evaluates to false for
 all the elements of the list.

**Generic parameters**
  * `'T` is `'b`
```

### Preview from Ionide

**Before**
![2018-07-21 10 59 51](https://user-images.githubusercontent.com/4760796/43034102-30c7565a-8cd6-11e8-9db3-f98a12a5a2bf.gif)

**Now**
![2018-07-20 23 17 05](https://user-images.githubusercontent.com/4760796/43034107-3a47895c-8cd6-11e8-992f-4dbec2380d4c.gif)

The new preview use the same section as docs comment in markdown style or XML style written by user.

Commit:
Improve tooltip text format

Krzysztof-Cieslak commented:
If we could keep changes into the functions used only by `formatTipEnhanced` and not by `formatTip` I'd be OK to merge it without any problems. This probably would required moving changes to `.ToString()` override to it's own method. And making sure that `formatTipEnhanced` uses it. 

It may require bit bigger refactoring of `TipFormater.fs`

MangelMaxime commented:
Ok, I see how to change the code according to your comments.

Just one question, why do we have a "classic" and an "enhanced" version ? Which one is use when ? Just so I can also test in Ionide the display of the classic one if we retrieve them somewhere.

Krzysztof-Cieslak commented:
"classic" is used by other editors, "enhanced" is used by Ionide

MangelMaxime commented:
Oh, so this solve my question about other editors support. perfect :)

Commit:
Separate classic and enhanced tip formatter

Classic is used by others editor than Ionide and return free text format

Enchanced is used by Ionide and will return markdown

MangelMaxime commented:
Seems like test `NotRestoredRunner.fsx` is stuck and never complete...

But I have no idea, what broke it.

Krzysztof-Cieslak commented:
Pretty sure it was broken before. LGTM
